stateserver.py

```diff
@@ -389,11 +389,13 @@ def post_wsSessions() -> Any:
             if currentShow and currentShow["connid"] == conn["connid"]:
                 # they should be on air now, but they've just died. go to jukebox.
                 # but don't kill it during the news, or after the end time, to avoid unexpected jukeboxing
-                now = datetime.datetime.now().timestamp()
-                if now < (currentShow["endTimestamp"] - 15):
-                    print("jukeboxing due to their disappearance...")
-                    subprocess.Popen(['sel', str(SOURCE_JUKEBOX)])
-                    do_ws_srv_telnet("NUL")
+                # Also, avoid killing them if they're on a non-WS source
+                if currentShow["sourceid"] != 5:
```

Comment:
I'm not 100% sure if sourceid is an int or a string...

Reply:
Couldn't you just force currentShow["sourceid"] to be an int?

Reply:
https://github.com/UniversityRadioYork/WebStudio/blob/markspolakovs-dont-kill-ob-line/stateserver.py#L241 leads me to believe it's an int, I'll take my chances